src/BlobStore.hs

```diff
@@ -56,62 +59,54 @@ import Test.QuickCheck.Instances.ByteString ()
 blobNameLength :: Int
 blobNameLength = 32
 
-newtype Password =
-  Pass BS.ByteString
-  deriving (Show)
+newtype Password = Pass BS.ByteString deriving (Show)
 
 -- HMAC specifies that keys less than the key length are null-padded up
 -- to the key length (longer keys are hashed).  This means that keys that
 -- are both less than the key length and differ only by trailing \NULs are
 -- equivalent HMAC keys.  We must surface this minor quirk here so that when
 -- we ask quickcheck for two different passwords, they actually differ.
 instance Eq Password where
-  (Pass a) == (Pass b)
+  Pass a == Pass b
     | BS.length a <= blobNameLength && BS.length b <= blobNameLength =
-      nullPad a == nullPad b
+        nullPad a == nullPad b
     | otherwise = a == b
     where
-      nullPad s = BS.append s (BS.replicate (blobNameLength - BS.length s) 0)
+      nullPad s = BS.append s $ BS.replicate (blobNameLength - BS.length s) 0
 
 instance Arbitrary Password where
   arbitrary = Pass <$> arbitrary
 
-newtype ExtantBlobName =
-  ExtantBlob BS.ByteString
-  deriving (Eq, Ord)
+newtype ExtantBlobName = ExtantBlob BS.ByteString deriving (Eq, Ord)
 
-blobName ::
-     MonadResource m => Password -> ConduitT BS.ByteString o m BS.ByteString
-blobName (Pass password) = do
-  digest <- sinkHMAC password
-  pure $ BA.convert (digest :: HMAC Crypto.Hash.Algorithms.SHA256)
+blobName
+  :: MonadResource m
+  => Password -> ConduitT BS.ByteString o m BS.ByteString
+blobName (Pass password) = BA.convert @(HMAC SHA256) <$> sinkHMAC password
 
-data StagedBlobHandle =
-  StagedBlobHandle
-    { commit :: BS.ByteString -> IO ExtantBlobName
-    , abort :: IO ()
-    }
+data StagedBlobHandle = StagedBlobHandle
+  { commit :: BS.ByteString -> IO ExtantBlobName
+  , abort :: IO ()
+  }
 
 class BlobStore a where
-  sinkBlob ::
-       MonadResource m => a -> ConduitT BS.ByteString o m StagedBlobHandle
+  sinkBlob
+    :: MonadResource m
+    => a -> ConduitT BS.ByteString o m StagedBlobHandle
   listBlobs :: a -> IO [ExtantBlobName]
   getBlob :: a -> ExtantBlobName -> ConduitT i BS.ByteString (ResourceT IO) ()
 
-sinkNamePrefixedBlob ::
-     (BlobStore bs, MonadResource m)
-  => bs
-  -> Password
-  -> ConduitT BS.ByteString o m (Maybe ExtantBlobName)
+sinkNamePrefixedBlob
+  :: (BlobStore bs, MonadResource m)
+  => bs -> Password -> ConduitT BS.ByteString o m (Maybe ExtantBlobName)
 sinkNamePrefixedBlob bs password = do
   name <- takeExactlyE blobNameLength await
   case name of
-    Just aname
-      | BS.length aname == blobNameLength -> sinkUntrustedBlob bs password aname
+    Just aname | BS.length aname == blobNameLength ->
+      sinkUntrustedBlob bs password aname
```

Comment:
I think it's easier to read here with the pattern match and condition on the same line.